lib/saluki-components/src/destinations/datadog_metrics/request_builder.rs

```diff
@@ -243,11 +256,79 @@ where
 
         self.compression_estimator.reset();
 
+        // Clear scratch buf data.
+        let scratch_buf = std::mem::take(&mut self.scratch_buf);
```

Comment:
Taking the existing scratch buffer requires that we replace it with a new `Vec<u8>`, which means we're always going to be allocating a new scratch buffer every time we flush.

I would probably just add a new method, `clear_scratch_buffer`, or something... and have it clear the buffer, and the lengths buffer. You'd then add a call to `clear_scratch_buffer` in the relevant spots:

- before we return if we fail to shutdown the compressor (right before line 266)
- right before we return the single request in the non-split scenario (right before line 279)
- right before we return our split requests (right before line 324)